Frontend/src/Rewards.jsx

```diff
@@ -1,35 +1,66 @@
-import {  } from "react";
+import { useState, useEffect } from "react";
 import PropTypes from "prop-types";
+import "./css/Rewards.css"
+import gem from "../src/img/gem.png"
 
-export default function Rewards({courses, userData}) {
+export default function Rewards({courses, setOverlay}) {
+  const [subArr, setSubArr] = useState([])
+  const [index, setIndex] = useState(0)
 
-  const newCompletions = courses.map((c) => {
-    c[2].filter((a) => {a.length > 3 && Date.parse(a[4]) > userData.lastLogin})
-  })
-  console.log(newCompletions)
+  useEffect(() => {
+    var subs = []
 
-  // return (
-  //   newCompletions.map((c) => {
-  //     c ? c.map((a) => {return <div key={a}>hi</div>}) : {}
-  //   })
-  // )
+    courses.forEach((c) => {
+      if (c[3].length > 0) {
+        subs.push([c[1], c[3]]);
+      }
+    })
+
+    console.log("subs", subs)
+
+    if (subs.length == 0) {
+      setOverlay("home")
+    } else {
+      setSubArr(subs)
+    }
+  }, [courses])
```

Comment:
I think `subArr` doesn't need to be a state variable updated in a use effect, it can just be some `const` variable based on courses (or memo if this isn't fast). I'll look into this for the future though because of time constraints